tests/Integration/PageFrontendControllerTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types=1);
+
+namespace IntegerNet\GlobalCustomLayout\Test\Integration;
+
+use IntegerNet\GlobalCustomLayout\Test\Util\CustomLayoutManager;
+use IntegerNet\GlobalCustomLayout\Test\Util\PageLayoutUpdateManager;
+use Magento\Cms\Api\Data\PageInterface;
+use Magento\Cms\Model\Page as PageModel;
+use Magento\Cms\Model\Page\CustomLayoutRepositoryInterface;
+use Magento\Cms\Model\PageFactory as PageModelFactory;
+use Magento\Cms\Model\ResourceModel\Page as PageResourceModel;
+use Magento\Framework\Exception\AlreadyExistsException;
+
+class PageFrontendControllerTest extends AbstractFrontendControllerTest
+{
+    /** @var CustomLayoutManager */
+    protected $layoutManager;
+
+    /** @var PageModel */
+    protected $page;
+
+    /** @var PageResourceModel */
+    protected $pageResource;
+
+    /** @var pageFactory $pageFactory */
+    protected $pageFactory;
+
+    protected function getPage(int $storeId, string $pageIdentifier): pageModel
```

Comment:
`getPage` implies that this is a query method that retrieves something that already exists. You call `load()` here, so it seems to make sense. But where does the page come from? Isn't it created here in the first place because it doesn't exist yet in the DB? 
Then `createPage` would be more appropiate, and instead of `load` something like:
```
$page->setId($pageIdentifier);
```

Reply:
Also, if you create entitites in the DB during tests, either make sure to delete them at the end, or use `@magentoDbIsolation enabled` in the test

Reply:
The page is created through `@magentoDataFixture Magento/Cms/_files/pages.php`.
So, it's cleaned up afterwards by `pages_rollback.php`.

Is it bad to rely on core fixtures?

In core they save the _selected custom layout file_ in the fixture, which means that code is not covered by the test, which I disliked.

I could still write my own fixture to create and destroy a product, but I'd still save the _global selected layout_ in my test.

Reply:
Ah, I got it. It can be convenient to rely on core fixtures if they provide what you need, especially in standalone extensions. But as you see by my reaction, they can be confusing (this is a case of a [Mystery Guest](https://www.integer-net.com/test-smell-mystery-guest/)

It would help here, to make `page100`  a constant `PAGE_ID_FROM_FIXTURE`

Reply:
Nice! I should spend some time on our blog :D

Reply:
Resolved. Please check again :)